modules/jfx.incubator.richtext/src/main/java/jfx/incubator/scene/control/richtext/CodeArea.java

```diff
@@ -417,16 +419,19 @@ public StyleableProperty<Number> getStyleableProperty(CodeArea n) {
      * @return plain text
      */
     public final String getText() {
-        // TODO or use save(DataFormat, Writer) ?
         StyledTextModel m = getModel();
-        StringBuilder sb = new StringBuilder(4096);
-        int sz = m.size();
-        for(int i=0; i<sz; i++) {
-            String s = m.getPlainText(i);
-            sb.append(s);
-            sb.append('\n');
+        if (m == null) {
+            return null;
+        }
+        TextPos end = m.getDocumentEnd();
+        try (StringBuilderStyledOutput out = new StringBuilderStyledOutput()) {
+            out.setLineSeparator("\n");
+            m.export(TextPos.ZERO, end, out);
+            return out.toString();
+        } catch (IOException e) {
+            // should not happen
+            throw new RuntimeException(e);
```

Comment:
Minor: wrapping it in `UncheckedIOException` is preferred over raw `RuntimeException`.